index/scorch/persister.go

```diff
@@ -142,21 +146,38 @@ OUTER:
 				}
 				close(ch)
 			}
+
```

Comment:
To fire a callback,
What if we just move the reset of the s.persistedCallbacks from line no: 133 to 181 or so.
(Reset it only on a success case) and fire the callback wherever needed than having all these newer temp slices?